### DISCUSSION TITLE
Checking if data is from optimistic update

### DISCUSSION THREAD

**Discussion by andresespinosapc** [TanStack/query#7557] [Q&A]
I have a checkout in my app. In almost all parts of my app I want the price to be updated optimistically, because users can change parameters that change the price and they should be able to compare prices quickly.

When the user is gonna press the "Buy" button I want to show the price that is saved in the server, not the optimistic one, so they don't buy something different by mistake.

How do you think I should solve this?

**Comment by serdar-webdev** [comment]
you can use optimistic updates for your prices and after clicking buy a product, you can implement refetching your query, and after that starting your mutation aka buying a product
